lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -5315,6 +5316,87 @@ class DecomposeAtenStdDimOp : public OpRewritePattern<AtenStdDimOp> {
 };
 } // namespace
 
+// Decompose aten.rot90
+// github:
+// https://github.com/pytorch/pytorch/blob/207564bab1c4fe42750931765734ee604032fb69/torch/_refs/__init__.py#L3830
+namespace {
+class DecomposeAtenRot90Op : public OpRewritePattern<AtenRot90Op> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenRot90Op op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.getSelf();
+
+    // Convert dims from Value to SmallVector.
+    SmallVector<Value> dims;
+    if (!getListConstructElements(op.getDims(), dims))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: dims not list of Scalar");
+
+    // Define constants.
+    Value cstZero = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(0));
+
+    int64_t dim0;
+    if (!matchPattern(dims[0], m_TorchConstantInt(&dim0)))
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented: dims[0] not constant int");
+    Value cstDim0 = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(dim0));
+
+    int64_t dim1;
+    if (!matchPattern(dims[1], m_TorchConstantInt(&dim1)))
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented: dims[0] not constant int");
+    Value cstDim1 = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(dim1));
```

Comment:
Is this really required? We're extracting the integer value out of the `TorchConstantInt` and then again creating a `TorchConstantInt` with that integer value.